src/Orleans/Telemetry/Consumers/FileTelemetryConsumer.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.IO;
+
+namespace Orleans.Runtime
+{
+    public class FileTelemetryConsumer : ITraceTelemetryConsumer
+    {
+        private StreamWriter _logOutput;
+        private readonly object _lockObj = new object();
+
+        public FileTelemetryConsumer(string fileName) : this(new FileInfo(fileName))
+        {
+        }
+
+        public FileTelemetryConsumer(FileInfo file)
+        {
+            var fileExists = File.Exists(file.FullName);
+            _logOutput = fileExists ? file.AppendText() : file.CreateText();
+            file.Refresh();
+        }
+
+        public void TrackTrace(string message)
+        {
+            lock (_lockObj)
+            {
+                if (_logOutput == null) return;
+                _logOutput.WriteLine(message);
+                _logOutput.Flush();
```

Comment:
I would keep it as is, with manual flush (and the lock). Both issues are unrelated to this PR. Open another PR with suggested changes to how `LogWriterToFile` works.